livekit-agents/livekit/agents/transcription/stt_forwarder.py

```diff
@@ -0,0 +1,87 @@
+import asyncio
+import contextlib
+import uuid
+
+from livekit import rtc
+
+from .. import stt
+from ..log import logger
+
+
+def _uuid() -> str:
+    return str(uuid.uuid4())[:12]
+
+
+class STTSegmentsForwarder:
+    """
+    Forward the STT transcription and keep the right timing info
+    """
+
+    def __init__(
+        self,
+        *,
+        room: rtc.Room,
+        participant_identity: str,
+        track_id: str,
+    ):
+        self._room = room
+        self._participant_identity = participant_identity
+        self._track_id = track_id
+        self._queue = asyncio.Queue[rtc.TranscriptionSegment | None]()
+        self._main_task = asyncio.create_task(self._run())
+        self._current_id = _uuid()
+
+    async def _run(self):
+        try:
+            while True:
+                seg = await self._queue.get()
+                if seg is None:
+                    break
+
+                transcription = rtc.Transcription(
```

Comment:
should there be rate limiting (i.e. max once a second) and bundling? where if there are multiple updates queued for the same segment_id, it feels like we shouldn't send multiple updates.. (TODO is fine too)

Reply:
I don't think STT often change the segment, I think it is OK to forward everything we get from them